src/extras/render-passes/render-pass-depth-aware-blur.js

```diff
@@ -1,93 +1,197 @@
+            super.execute();
+            return;
+        }
 
-            float random(const highp vec2 w) {
-                const vec3 m = vec3(0.06711056, 0.00583715, 52.9829189);
-                return fract(m.z * fract(dot(w, m.xy)));
-            }
+        const { width, height } = this.sourceTexture;
+        sourceInvResolutionValueTmp[0] = 1.0 / width;
+        sourceInvResolutionValueTmp[1] = 1.0 / height;
 
-            float bilateralWeight(in float depth, in float sampleDepth) {
-                float diff = (sampleDepth - depth);
-                return max(0.0, 1.0 - diff * diff);
-            }
+        this.sourceInvResolutionId?.setValue(sourceInvResolutionValueTmp);
+        this.sourceTextureId?.setValue(this.sourceTexture);
+        this.kernelId?.setValue(this.kernelValue);
+        this.directionId?.setValue(this.direction);
 
-            void tap(inout float sum, inout float totalWeight, float weight, float depth, vec2 position) {
+        super.execute();
+    }
 
-                float color = texture2D(sourceTexture, position).r;
-                float textureDepth = -getLinearScreenDepth(position);
-            
-                float bilateral = bilateralWeight(depth, textureDepth);
+    /**
+     * @param {BLUR_GAUSSIAN|import('../../scene/constants.js').BLUR_BOX} value - The type of blur to apply.
+     */
+    set type(value) {
+        if (value === this._type) {
+            return;
+        }
 
-                bilateral *= weight;
-                sum += color * bilateral;
-                totalWeight += bilateral;
-            }
+        this._type = value;
 
-            // TODO: weights of 1 are used for all samples. Test with gaussian weights
-            void main() {
+        this.updateWeightCoefs();
+    }
 
-                // handle the center pixel separately because it doesn't participate in bilateral filtering
-                float depth = -getLinearScreenDepth(uv0);
-                float totalWeight = 1.0;
-                float color = texture2D(sourceTexture, uv0 ).r;
-                float sum = color * totalWeight;
+    get type() {
+        return this._type;
+    }
 
-                for (int x = -filterSize; x <= filterSize; x++) {
-                   for (int y = -filterSize; y < filterSize; y++) {
-                       float weight = 1.0;
-                       vec2 offset = vec2(x,y) * sourceInvResolution;
-                       tap(sum, totalWeight, weight, depth, uv0 + offset);
-                   }
-                }
+    set kernelSize(value) {
+        if (value === this._kernelSize) {
+            return;
+        }
 
-                float ao = sum / totalWeight;
+        this._kernelSize = Math.max(1, Math.floor(value));
 
-                // simple dithering helps a lot (assumes 8 bits target)
-                // this is most useful with high quality/large blurs
-                // ao += ((random(gl_FragCoord.xy) - 0.5) / 255.0);
+        this.updateShader();
```

Comment:
ideally don't call updateShader directly, just set a _shaderDirty flag - this makes it easy to add more properties that modify the shader, and also better handle the case where the property is changed multi times per frame. (not that this is typical, but happens).